exporter/kinesisexporter/marshaller.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kinesisexporter
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// Marshaller marshals traces/metrics into Message array.
+type Marshaller interface {
+	// MarshalMetrics serializes metrics into Messages
+	MarshalMetrics(metrics pdata.Metrics) ([]Message, error)
+
+	// Encoding returns encoding name
+	Encoding() string
+}
+
+// Message encapsulates Kafka's message payload.
+type Message struct {
```

Comment:
Do we need this extra struct wrapper around the byte array?

Reply:
I was thinking this will include the partition key since by the time we move it to bytes we've no access to the underlying metrics data

Reply:
Though looking at the otlp kafka exporter implementation partition key isn't necessarily specified so we can skip that